dotnet/src/Microsoft.Agents.AI.Hosting.OpenAI/OpenAIHostingJsonUtilities.cs

```diff
@@ -4,16 +4,62 @@
 using System.Diagnostics.CodeAnalysis;
 using System.Text.Json;
 using System.Text.Json.Serialization;
+using Microsoft.Agents.AI.Hosting.OpenAI.Conversations.Models;
+using Microsoft.Agents.AI.Hosting.OpenAI.Models;
 using Microsoft.Agents.AI.Hosting.OpenAI.Responses.Models;
+using Microsoft.Extensions.AI;
 
-namespace Microsoft.Agents.AI.Hosting.OpenAI.Responses;
+namespace Microsoft.Agents.AI.Hosting.OpenAI;
 
+/// <summary>
+/// Provides JSON serialization options and context for OpenAI Hosting APIs to support AOT and trimming.
+/// </summary>
+internal static class OpenAIHostingJsonUtilities
+{
+    /// <summary>
+    /// Gets the default <see cref="JsonSerializerOptions"/> instance used for OpenAI API serialization.
+    /// Includes support for AIContent types and all OpenAI-related types.
+    /// </summary>
+    public static JsonSerializerOptions DefaultOptions { get; } = CreateDefaultOptions();
+
+    private static JsonSerializerOptions CreateDefaultOptions()
+    {
+        // Start with our source-generated context
+        JsonSerializerOptions options = new(OpenAIHostingJsonContext.Default.Options);
+
+        // Chain with agent abstraction types
+        options.TypeInfoResolverChain.Add(AgentAbstractionsJsonUtilities.DefaultOptions.TypeInfoResolver!);
+
+        // Chain with AIContent types from Microsoft.Extensions.AI
+        options.TypeInfoResolverChain.Add(AIJsonUtilities.DefaultOptions.TypeInfoResolver!);
```

Comment:
This shouldn't be necessary, as AgentAbstractionsJsonUtilities already chains it in.